packages/framework/ai-collab/src/explicit-strategy/promptGeneration.ts

```diff
@@ -130,20 +134,21 @@ export function getEditingSystemPrompt(
 	const role = `You are a collaborative agent who interacts with a JSON tree by performing edits to achieve a user-specified goal.${
 		appGuidance === undefined
 			? ""
-			: `
-			The application that owns the JSON tree has the following guidance about your role: ${appGuidance}`
+			: `\nThe application that owns the JSON tree has the following guidance about your role: "${appGuidance}".`
 	}`;
 
 	const treeSchemaString = createZodJsonValidator(
 		...generateGenericEditTypes(getSimpleSchema(schema), false),
 	).getSchemaText();
 
+	const topLevelEditWrapperDescription = doesNodeContainArraySchema(treeNode)
+		? `contains one of the following interfaces: "Insert", "Modify", "Remove", "Move", or null`
+		: `contains the interface "Modify" or null`;
+
 	// TODO: security: user prompt in system prompt
 	const systemPrompt = `
-	${role}
-	Edits are JSON objects that conform to the following schema.
-	The top level object you produce is an "EditWrapper" object which contains one of "Insert", "Modify", "Remove", "Move", or null.
-	${treeSchemaString}
+	${role}\nEdits are JSON objects that conform to the following schema of which the top level object you produce is an "EditWrapper" object which ${topLevelEditWrapperDescription}:
```

Comment:
Nit: there's a bit of run-on sentence feeling here (especially the double "which"), something more like the previous version reads clearer. What do you think of `"Edits are JSON objects that conform to the schema described below. The top-level object you produce for a given edit is an `EditWrapper` object which [...]. This is the schema description for an edit:\n{treeSchemaString}"`?

And similar to the comment above, wondering if it would help if the text where we include the array-related operations explicitly says that those only apply in the context of an array node.

Reply:
Agree that the sentence is weird here. I've updated it in commit c595f44eb33065af14eb1b92eccbf7eb4b35ea09 to be
```
Edits are JSON objects that conform to the schema described below. The top-level object you produce for a given edit is an "EditWrapper" object which contains one of the following interfaces: "Modify", null or an array node only edit: "Insert", "Remove", "Move"..
	
Here are the schema definitions for an edit:
```

Reply:
Also, note I removed the extra trailing period at the end of the sentence I introduced in commit c595f44eb33065af14eb1b92eccbf7eb4b35ea09 in a follow up commit: b0c518262d8226ea34c595a87028cd898aa036df